### PR TITLE
[RELENG-571] check existing privileged add-on IDs against allow list

### PR DESCRIPTION
**Pull request by escapewindow**
Also remove news-disinformation-study, which doesn't match the allowlist.

### CODE REVIEW

**Comment by escapewindow** [comment]
@wagnerand , it looks like news-disinformation-study doesn't match:

```
[task 2021-09-09T01:19:42.510Z] Exception: applications.gecko.id news.study@princeton.edu must end with one of the following suffixes!
```

Should we EOL that addon? change its ID? allowlist just that one ID? Something else?

**Comment by escapewindow** [comment]
Blocked until the 13th, unless @kewisch knows the answer.

**Comment by jmaher** [comment]
there is a sunsetting plan for the news-disinformation-study addon, maybe @rhelmer could shed some light

**Review by jmaher — APPROVED** [review]
this looks good.  the error messages are useful which makes me happy.

**Review by wagnerand — APPROVED** [review]
Thank you for tackling this, looks good to me!

Maybe @diox could have a quick look as well.

File: taskcluster/docker/node/build.py [hunk 1]
```diff
@@ -65,6 +81,30 @@ def get_hash(path, hash_alg="sha256"):
...
+        try:
+            with xpi.open(manifest_name, "r") as f:
+                manifest = json.load(f)
+        except KeyError:
+            print(f"{manifest_name} doesn't exist in {path}...")
+            continue
+        for _key in ("applications", "browser_specific_settings"):
+            _id = manifest.get(_key, {}).get("gecko", {}).get("id", None)
+            if _id is None:
+                continue
```
Comment (diox) [inline taskcluster/docker/node/build.py:98]: Can this happen with add-ons handled by `shipit` currently ?

Comment (escapewindow) [reply taskcluster/docker/node/build.py:98]: I think we'll break in the signing step. This block would catch it earlier, in the build step.

**Review by diox — APPROVED** [review]

**Comment by escapewindow** [comment]
Thanks!

**Pull request merged**